@pbaughman I found some other unrelated failures.  Am I using it wrong?
(I `source /opt/ApexOS/install.bash`)
```
ade$ colcon test --packages-select apex_launchtest --event-handler console_direct+
...

=================================== FAILURES ===================================
_______________________ test_examples[good_proc.test.py] _______________________
test/test_examples.py:45: in test_examples
    assert 0 == subprocess.run(args=proc).returncode
E   AssertionError: assert 0 == 1
E    +  where 1 = CompletedProcess(args=['apex_launchtest', '/tmp/apex_rostest/install/share/apex_launchtest/examples/good_proc.test.py'], returncode=1).returncode
E    +    where CompletedProcess(args=['apex_launchtest', '/tmp/apex_rostest/install/share/apex_launchtest/examples/good_proc.test.py'], returncode=1) = <function run at 0x7fdf807a40d0>(args=['apex_launchtest', '/tmp/apex_rostest/install/share/apex_launchtest/examples/good_proc.test.py'])
E    +      where <function run at 0x7fdf807a40d0> = subprocess.run
----------------------------- Captured stderr call -----------------------------
Traceback (most recent call last):
  File "/tmp/apex_rostest/install/bin/apex_launchtest", line 3, in <module>
    from apex_launchtest.apex_launchtest_main import apex_launchtest_main
  File "/tmp/apex_rostest/build/apex_launchtest/apex_launchtest/apex_launchtest_main.py", line 21, in <module>
    from ros2launch.api import print_arguments_of_launch_description
ImportError: cannot import name 'print_arguments_of_launch_description'
_________________________ test_examples[args.test.py] __________________________
test/test_examples.py:45: in test_examples
    assert 0 == subprocess.run(args=proc).returncode
E   AssertionError: assert 0 == 1
E    +  where 1 = CompletedProcess(args=['apex_launchtest', '/tmp/apex_rostest/install/share/apex_launchtest/examples/args.test.py', 'dut_arg:=foobarbaz'], returncode=1).returncode
E    +    where CompletedProcess(args=['apex_launchtest', '/tmp/apex_rostest/install/share/apex_launchtest/examples/args.test.py', 'dut_arg:=foobarbaz'], returncode=1) = <function run at 0x7fdf807a40d0>(args=['apex_launchtest', '/tmp/apex_rostest/install/share/apex_launchtest/examples/args.test.py', 'dut_arg:=foobarbaz'])
E    +      where <function run at 0x7fdf807a40d0> = subprocess.run
----------------------------- Captured stderr call -----------------------------
Traceback (most recent call last):
  File "/tmp/apex_rostest/install/bin/apex_launchtest", line 3, in <module>
    from apex_launchtest.apex_launchtest_main import apex_launchtest_main
  File "/tmp/apex_rostest/build/apex_launchtest/apex_launchtest/apex_launchtest_main.py", line 21, in <module>
    from ros2launch.api import print_arguments_of_launch_description
ImportError: cannot import name 'print_arguments_of_launch_description'
---- generated xml file: /tmp/apex_rostest/build/apex_launchtest/pytest.xml ----
===================== 2 failed, 34 passed in 17.54 seconds =====================
Finished <<< apex_launchtest [18.5s]    [ with test failures ]

Summary: 1 package finished [18.8s]
  1 package had test failures: apex_launchtest
```

--- a/apex_launchtest/test/test_xml_output.py
+++ b/apex_launchtest/test/test_xml_output.py
@@ -1,0 +1,60 @@
+# Copyright 2019 Apex.AI, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import subprocess
+import tempfile
+import unittest
+import xml.etree.ElementTree as ET
+
+import ament_index_python
+
+
+class TestGoodXmlOutput(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        # For performance, we'll run the test once and generate the XML output, then
+        # have multiple test cases assert on it
+
+        cls.tmpdir = tempfile.TemporaryDirectory()
+        cls.xml_file = os.path.join(cls.tmpdir.name, 'junit.xml')
+
+        path = os.path.join(
+            ament_index_python.get_package_share_directory('apex_launchtest'),
+            'examples',
+            'good_proc.test.py'
+        )
+
+        assert 0 == subprocess.run(
+            args=[
+                'apex_launchtest',
+                path,
+                '--junit-xml', os.path.join(cls.tmpdir.name, 'junit.xml'),
+            ],
+        ).returncode
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.tmpdir.cleanup()
+
+    def test_pre_and_post(self):
+        tree = ET.parse(self.xml_file)
+        root = tree.getroot()
+
+        self.assertEqual(len(root.getchildren()), 2)
+
+        # Expecting an element called "active_tests" and "after_shutdown_tests"
+        child_names = [chld.attrib['name'] for chld in root.getchildren()]
+        self.assertEqual(set(child_names), set(['active_tests', 'after_shutdown_tests']))

